contracts/secret-storage/SecretManager.sol

```diff
@@ -595,6 +596,72 @@ contract SecretManager is
         _replaceStore(_secretId, _enclaveAddress, enclaveIndex, true);
     }
 
+    function _renounceSecrets(
+        address _enclaveAddress
+    ) internal {
+        ( , address owner, , bool draining) = TEE_MANAGER.teeNodes(_enclaveAddress);
+        if(!draining)
+            revert SecretManagerStoreNotDraining();
+        // can use signature verification as well
+        if(_msgSender() != owner)
+            revert SecretManagerInvalidSecretOwner();
+
+        uint256 lastAliveTimestamp = SECRET_STORE.getSecretStoreLastAliveTimestamp(_enclaveAddress);
+        SECRET_STORE.renounceSecretsPreUpdate(_enclaveAddress, MARK_ALIVE_TIMEOUT, STAKING_PAYMENT_POOL);
+
+        // TODO: how to handle the newly assigned unacknowledged secrets
```

Comment:
I think we can prevent ack acceptance from the draining enclaves, and after sometime ack timeout will happen which will reassign or terminate the secret.

Reply:
added a check to prevent ack acceptance if the node is draining